### DISCUSSION TITLE
[Next.js] No cache-control headers in any of next.config.js or vercel.json is recognized

### DISCUSSION THREAD

**Discussion by axross** [vercel/vercel#6017] [Help]
Hi community, I quickly played around adding `cache-control` headers in my personal website made of Next.js.

As far as I attempted to set `cache-control` header in both/either of `next.config.js` and/or `vercel.json` and deployed to Vercel, none of them worked. The endpoints on Vercel always returned `cache-control: private, no-cache, no-store, max-age=0, must-revalidate` regardless what specified in those config files. However, it worked with no problem at my local environment.

Does anyone know some context or solution for this problem?

**next.config.js**

```js
module.exports = (phase) => {
  return {
    // ...
    headers: async () => [
      {
        source: "/",
        headers: [
          {
            key: "cache-control",
            value: "max-age=3600, stale-while-revalidate=86400, public",
          },
        ],
      },
      {
        source: "/posts/:slug",
        headers: [
          {
            key: "cache-control",
            value: "max-age=3600, stale-while-revalidate=86400, public",
          },
        ],
      },
    ],
  };
};
```

**vercel.json**

```json
{
  "headers": [
    {
      "source": "/",
      "headers": [
        {
          "key": "cache-control",
          "value": "max-age=3600, stale-while-revalidate=86400, public"
        }
      ]
    },
    {
      "source": "/posts/:slug",
      "headers": [
        {
          "key": "cache-control",
          "value": "max-age=3600, stale-while-revalidate=86400, public"
        }
      ]
    }
  ]
}
```

Just for your information, there's workaround that you can manually set the header in `getServerSideProps()`:

```js
export async function getServerSideProps({ res }) {
  // WORKAROUND
  res.setHeader("cache-control", "max-age=3600, stale-while-revalidate=86400, public");

  // ...

  return {
    props: {
      // ...
    },
  };
}
```

ref: https://github.com/axross/kohei.dev/pull/234

**Comment by anhthien-bic** [comment]
same :(
